From: Arnaud Ferraris <arnaud.ferraris@collabora.com>
Date: Fri, 12 Jan 2024 14:20:44 +0100
Subject: drivers: touchscreen: fix conflict between downstream drivers

Both the `goodix_berlin_a` (FP5) and `hxchipset` (FP4) downstream
drivers declare non-static `check_dt()` and `check_default_tp()`
functions, although none of those is used outside of the file where
they're declared. This results in a build failure at link time.

Mark those as `static` in order to fix the build.
---
 drivers/input/touchscreen/goodix_berlin_a_driver/goodix_ts_core.c | 4 ++--
 drivers/input/touchscreen/hxchipset/himax_platform.c              | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/input/touchscreen/goodix_berlin_a_driver/goodix_ts_core.c b/drivers/input/touchscreen/goodix_berlin_a_driver/goodix_ts_core.c
index 7ee0592..5a356d3 100644
--- a/drivers/input/touchscreen/goodix_berlin_a_driver/goodix_ts_core.c
+++ b/drivers/input/touchscreen/goodix_berlin_a_driver/goodix_ts_core.c
@@ -251,7 +251,7 @@ static ssize_t goodix_ts_screen_mode_store(struct device *dev,
 #include <drm/drm_panel.h>
 struct drm_panel *gdix_active_panel;
 
-int check_dt(struct device_node *np)
+static int check_dt(struct device_node *np)
 {
 	int i;
 	int count;
@@ -276,7 +276,7 @@ int check_dt(struct device_node *np)
 	return -ENODEV;
 }
 
-int check_default_tp(struct device_node *dt, const char *prop)
+static int check_default_tp(struct device_node *dt, const char *prop)
 {
 	const char *active_tp;
 	const char *compatible;
diff --git a/drivers/input/touchscreen/hxchipset/himax_platform.c b/drivers/input/touchscreen/hxchipset/himax_platform.c
index 67dda8f..e811ea2 100644
--- a/drivers/input/touchscreen/hxchipset/himax_platform.c
+++ b/drivers/input/touchscreen/hxchipset/himax_platform.c
@@ -27,7 +27,7 @@ u8 *gp_rw_buf;
 #if defined(HX_CONFIG_DRM)
 struct drm_panel *active_panel;
 
-int check_dt(struct device_node *np)
+static int check_dt(struct device_node *np)
 {
 	int i;
 	int count;
@@ -52,7 +52,7 @@ int check_dt(struct device_node *np)
 }
 #endif
 
-int check_default_tp(struct device_node *dt, const char *prop)
+static int check_default_tp(struct device_node *dt, const char *prop)
 {
 	const char *active_tp;
 	const char *compatible;
